tests/snap_test.go

```diff
@@ -215,22 +227,35 @@ func TestWifiMatterCommander(t *testing.T) {
 	var stdout, stderr string
 	var err error
 
+	start := time.Now()
+
+	t.Cleanup(func() {
+		utils.SnapStop(t, snapMatterPiGPIO)
+
+		if env.Teardown() {
+			// Remove snaps, ignore errors during removal
+			utils.SnapRemove(nil, chipToolSnap)
+			// snapMatterPiGPIO is not removed here, as that is handled by the teardown function
+		}
+
+		utils.SnapDumpLogs(t, start, snapMatterPiGPIO)
+	})
+
 	// install chip-tool
-	err = utils.SnapInstallFromStore(t, chipToolSnap, utils.ServiceChannel)
+	err = utils.SnapInstallFromStore(t, chipToolSnap, "latest/beta")
 	if err != nil {
 		t.Fatalf("Failed to install chip-tool: %s", err)
 	}
-	t.Cleanup(func() { utils.SnapRemove(t, chipToolSnap) })
 
 	// chip-tool interfaces
 	utils.SnapConnect(nil, chipToolSnap+":avahi-observe", "")
 
 	utils.SnapStart(t, snapMatterPiGPIO)
-	t.Cleanup(func() { utils.SnapStop(t, snapMatterPiGPIO) })
 
 	// commission
 	t.Run("Commission", func(t *testing.T) {
 		stdout, stderr, err = utils.Exec(t, "sudo chip-tool pairing onnetwork 110 20202021")
+		assert.NoError(t, utils.WriteLogFile(t, chipToolSnap, stdout))
```

Comment:
```suggestion
                assert.NoError(t, err)
		assert.NoError(t, utils.WriteLogFile(t, chipToolSnap, stdout))
```
No introduced by this PR, but errors aren't checked, in several places.